src/protocols/LoRaWAN/LoRaWAN.cpp

```diff
@@ -697,17 +722,18 @@ int16_t LoRaWANNode::beginABP(uint32_t addr, uint8_t* nwkSKey, uint8_t* appSKey,
   if(fNwkSIntKey) { checkSum ^= LoRaWANNode::checkSum16(fNwkSIntKey, 16); }
   if(sNwkSIntKey) { checkSum ^= LoRaWANNode::checkSum16(sNwkSIntKey, 16); }
 
-  bool validCheckSum = mod->hal->getPersistentParameter<uint16_t>(RADIOLIB_EEPROM_LORAWAN_CHECKSUM_ID) == checkSum;
-  bool validMode = mod->hal->getPersistentParameter<uint16_t>(RADIOLIB_EEPROM_LORAWAN_MODE_ID) == RADIOLIB_LORAWAN_MODE_ABP;
+  bool isValidCheckSum = mod->hal->getPersistentParameter<uint16_t>(RADIOLIB_EEPROM_LORAWAN_CHECKSUM_ID) == checkSum;
+  bool isValidMode = mod->hal->getPersistentParameter<uint16_t>(RADIOLIB_EEPROM_LORAWAN_MODE_ID) == RADIOLIB_LORAWAN_MODE_OTAA;
 
-  if(!force && validCheckSum && validMode) {
-    // the device has joined already, we can just pull the data from persistent storage
-    RADIOLIB_DEBUG_PRINTLN("Found existing session; restoring...");
-    
+  if(isValidCheckSum && isValidMode && !force && this->isActiveSession()) {
```

Comment:
I think this logic can be simplified (I had a bit of a rough time trying to follow the current one). Since `isValidCheckSum` and `isValidMode` are not evaluated independently, it could be the following:

```c++
if(isValidCheckSum & isValidMode) {
  if(!force && this->isActiveSession) {
    return(this->restore());
  }
  // either forced or no active session
  this->clearSession();
} else {
  // either invalid checksum or mode, do the debug and wipe either way
  this->wipe();
}
```

Reply:
A similar point also applies to a similar section in beginOTAA